source/s3_client.c

```diff
@@ -649,10 +674,11 @@ struct aws_s3_meta_request *aws_s3_client_make_meta_request(
             .ref_count_zero_callback = client->vtable->endpoint_ref_count_zero,
             .shutdown_callback = client->vtable->endpoint_shutdown_callback,
             .client_bootstrap = client->client_bootstrap,
-            .tls_connection_options = client->tls_connection_options,
+            .tls_connection_options = is_https ? client->tls_connection_options : NULL,
```

Comment:
I think you requested to expose the client config to Java, right? That's totally reasonable. 

For different meta request, we can use different endpoint to connect, so, it makes sense to override the tls options from the client. 
We can probably add customized tls when needed.
I think that's fine, but we need to document it out.

Reply:
I have tested with custom TLS context against Minio, and built-in TLS against <my-bucket>.us-east-1.s3.amazonaws.com

Reply:
Just to clarify - I think you are recommending that the `TlsContext` be supported in the `S3MetaRequestOptions` and then each meta request can use a different TLS Context. Is that correct? If so I will modify the PR. (Result would be an additional arg `tlsContext` to the JNI `s3ClientMakeMetaRequest` method.)

Reply:
Nope, I think for now, it's fine. But, if anyone requests to add customized TLS for each meta request, it's also reasonable. Let's not do it until anyone asks for it.